gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java

```diff
@@ -97,6 +98,11 @@ public final void runTask() {
     run();
   }
 
+  public void addEffectiveImageUserAndEffectivePlatformModeProperty() {
+    // Should be overridden by child classes in case properties need
+    // to be added before resolving images
+  }
+
```

Comment:
2 notes:
- Why are you making this public? if its meant to be overridden by child classes, make it protected.
- Why do we need an extra method (so specific) when child classes can simply override the `initImageConfiguration`?

Reply:
I wanted to set `jkube.image.user` property just before resolving final images. Earlier property was being set after resolving images.

I made this method `public` since I wanted to use it in OpenShiftResourceTask, with protected access it would only be available in KubernetesResourceTask

Right now in `runTask`, `resolvedImages` is initialized. I couldn't find a way to set properties before initializing resolvedImages. `runTask` is set final so it can't be overridden.

Reply:
> Right now in runTask, resolvedImages is initialized. I couldn't find a way to set properties before initializing resolvedImages. runTask is set final so it can't be overridden.

OK, I thought `initImageConfiguration` was part of the AbstractJKubeTask, it's a static import (hard to see in the GH interface).

Anyway, `addEffectiveImageUserAndEffectivePlatformModeProperty` is **too** specific.

Maybe having a `resolveImages` method that can be overridden might be more interesting.
Other alternatives could be afterInit/lateInit, configuration, etc. methods that are located within different places in the runTask method. However, this might be early abstraction and become a burden.

> I made this method public since I wanted to use it in OpenShiftResourceTask, with protected access it would only be available in KubernetesResourceTask

That's a misunderstanding from your side, it would be this case if it was package-private (no access modifier). Protected methods can always be overridden in subclasses (unless declared final somewhere along the chain).

Reply:
I've added the resolveImages method in AbstractJKubeTask and overriding this from OpenShiftResourceTask